File: test/extensions/language/functions/java/functions/structure_extern/StructureExternTest.java
package functions.structure_extern;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;

import zserio.runtime.io.BitBuffer;

public class StructureExternTest
{
    @Test
    public void getField()
    {
        final TestStructure testStructure = new TestStructure();
        testStructure.setField(FIELD);
        testStructure.setChild(new Child(CHILD_FIELD));
        assertEquals(FIELD, testStructure.funcGetField());
    }

    @Test
    public void getChildField()
    {
        final TestStructure testStructure = new TestStructure();
        testStructure.setField(FIELD);
        testStructure.setChild(new Child(CHILD_FIELD));
        assertEquals(CHILD_FIELD, testStructure.funcGetChildField());
    }

    private static final BitBuffer FIELD = new BitBuffer(new byte[] {(byte)0xAB, (byte)0xE0}, 11);
    private static final BitBuffer CHILD_FIELD = new BitBuffer(new byte[] {(byte)0xCA, (byte)0xFE}, 15);
}

File: test/extensions/language/functions/python/StructureExternTest.py
import zserio

import Functions


class StructureExternTest(Functions.TestCase):
    def testGetField(self):
        testStructure = self.api.TestStructure(self.FIELD, self.api.Child(self.CHILD_FIELD))
        self.assertEqual(self.FIELD, testStructure.get_field())

    def testGetChildField(self):
        testStructure = self.api.TestStructure(self.FIELD, self.api.Child(self.CHILD_FIELD))
        self.assertEqual(self.CHILD_FIELD, testStructure.get_child_field())

    FIELD = zserio.BitBuffer(bytes([0xAB, 0xE0]), 11)
    CHILD_FIELD = zserio.BitBuffer(bytes([0xCA, 0xFE]), 15)
